packages/gatsby-source-mongodb/src/gatsby-node.js

```diff
@@ -71,14 +71,14 @@ function createNodes(
         reject(err)
       }
 
-      documents.forEach(item => {
-        var id = item._id.toString()
+      documents.map(d => JSON.parse(JSON.stringify(d))).forEach(item => {
```

Comment:
can we do `JSON.parse(JSON.stringify(d))` only when needed?

Reply:
A possible alternative would be to go through every value and see if `constructor.name` matches one of the non-JSON BSON types, and then call `.toString()` on the value. https://docs.mongodb.com/manual/reference/bson-types/

Reply:
can we do introspection in mongodb to know which fields need that? `JSON.parse(JSON.stringify(d))` just seems pretty expensive, especially if someone doesn't use non-json types and won't get anything other than performance penalty from it

Reply:
perhaps a plugin option like `convertBson` which defaults to false (which only converts the `_id` field).